### PR TITLE
Eahw 589 fix error message focus bug

### PR DESCRIPTION
**Pull request by JWeston-HO**
This PR moves the focus for errors outside the errorSummary object. To avoid a clash with onblur for shift ending on next day functionality.

### CODE REVIEW

File: src/utils/common-utils/common-utils.spec.js [hunk 1]
```diff
@@ -17,5 +20,32 @@ describe('common-utils', () => {
       expect(result).not.toBe(input);
       expect(result).toEqual(input);
     });
+
+    it('summary error should have focus and scroll into view', () => {
```
Comment (laurieboyes) [inline src/utils/common-utils/common-utils.spec.js:24]: I feel like this is part of the `deepClone` describe block by accident 🤔

File: src/utils/common-utils/common-utils.js [hunk 2]
```diff
@@ -1,3 +1,14 @@
 export const deepCloneJson = (objectOrArray) => {
   return JSON.parse(JSON.stringify(objectOrArray));
 };
+
+export const focusErrors = (firstSummaryError) => {
+  if (firstSummaryError) {
+    firstSummaryError.scrollIntoView({
```
Comment (laurieboyes) [inline src/utils/common-utils/common-utils.js:7]: Wow I know it’s copy/pasted code, but TIL about element.scrollIntoView, this is great 😄

File: src/utils/common-utils/common-utils.spec.js (hunk 1 above)
Comment (laurieboyes) [reply src/utils/common-utils/common-utils.spec.js:24]: Also could you rephrase the test description so that the `it` makes sense as part of the sentence?

>it('should do this cool thing')

File: src/components/timecard/select-timecard-period-type/SelectTimecardPeriodType.jsx [hunk 3]
```diff
@@ -25,13 +27,22 @@ const SelectTimecardPeriodType = () => {
...
+    setNewTimeEntry,
+  } = useTimecardContext();
 
   const handleError = (errorFields) => {
     setSummaryErrors(errorFields);
   };
 
+  useEffect(() => {
+    focusErrors(document.getElementById('summary-error-0-message'));
+  }, [summaryErrors]);
```
Comment (laurieboyes) [inline src/components/timecard/select-timecard-period-type/SelectTimecardPeriodType.jsx:44]: I’m trying to think when we would we get an error summary as part of `SelectTimecardPeriodType` 🤔

Comment (JWeston-HO) [reply src/components/timecard/select-timecard-period-type/SelectTimecardPeriodType.jsx:44]: When the user doesn't select a type and clicks continue.

File: src/utils/common-utils/common-utils.spec.js (hunk 1 above)
Comment (JWeston-HO) [reply src/utils/common-utils/common-utils.spec.js]: Fixed

File: src/utils/common-utils/common-utils.spec.js [hunk 4]
```diff
@@ -18,4 +21,31 @@ describe('common-utils', () => {
       expect(result).toEqual(input);
     });
   });
+
+  it('should set the summary error to have focus and scroll into view', () => {
```
Comment (laurieboyes) [inline src/utils/common-utils/common-utils.spec.js:25]: Please can you put this in a `describe('focusErrors', () => {…})` block?

**Review by laurieboyes — CHANGES_REQUESTED** [review]

File: src/utils/common-utils/common-utils.spec.js (hunk 4 above)
Comment (JWeston-HO) [reply src/utils/common-utils/common-utils.spec.js]: Actioned

**Review by laurieboyes — DISMISSED** [review]

File: src/components/timecard/add-timecard-period/AddTimeCardPeriod.spec.jsx [hunk 5]
```diff
@@ -28,10 +28,12 @@ describe('AddTimeCardPeriod component', () => {
 
   it('should call setNewTimeEntry when the add button is clicked', async () => {
     const setNewTimeEntrySpy = jest.fn();
+    const setSummarryErrorsSpy = jest.fn();
```
Comment (holly-blues) [inline src/components/timecard/add-timecard-period/AddTimeCardPeriod.spec.jsx:31]: Typo here `setSummarryErrorsSpy`

File: src/components/timecard/edit-shift-hours/EditShiftHours.jsx [hunk 6]
```diff
@@ -45,6 +49,11 @@ const EditShiftHours = ({
     summaryErrors,
     setSummaryErrors,
   } = useTimecardContext();
+
+  useEffect(() => {
+    focusErrors(document.getElementById('error-summary-0-message'));
```
[the remark this answers is not in the document]
Comment (JWeston-HO) [reply src/components/timecard/edit-shift-hours/EditShiftHours.jsx]: Good spot this is now addressed. It is 'summary-error-0-message'

File: src/components/timecard/add-timecard-period/AddTimeCardPeriod.spec.jsx (hunk 5 above)
Comment (JWeston-HO) [reply src/components/timecard/add-timecard-period/AddTimeCardPeriod.spec.jsx]: fixed

**Review by holly-blues — APPROVED** [review]

**Review by laurieboyes — APPROVED** [review]
